feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/RotationUtil.kt

```diff
@@ -0,0 +1,118 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.feature.preview.ui
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.os.Build
+import android.view.Surface
+import android.view.WindowManager
+import android.view.WindowManager.LayoutParams.ROTATION_ANIMATION_JUMPCUT
+import android.view.WindowManager.LayoutParams.ROTATION_ANIMATION_SEAMLESS
+import androidx.activity.ComponentActivity
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.layout
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.ui.unit.Constraints
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.WindowInsetsControllerCompat
+
+/**
+ * As long as this composable is active, the window will go into immersive mode and prevents the
+ * rotation animation on configuration change. This will prevent the UI items from visually changing.
+ *
+ * When used in combination with a composable that renders the same UI in both landscape and portrait,
+ * it can create a smooth continuous feel between those two orientations.
+ */
+@Composable
+fun SmoothImmersiveRotationEffect(context: Context) = DisposableEffect(context) {
+    var currentRotationAnimation: Int? = null
+    context.getActivity()?.window?.let { window ->
+        window.attributes = window.attributes.apply {
+            currentRotationAnimation = rotationAnimation
+            rotationAnimation = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                ROTATION_ANIMATION_SEAMLESS
+            } else {
+                ROTATION_ANIMATION_JUMPCUT
+            }
+        }
+        WindowCompat.getInsetsController(window, window.decorView).apply {
+            systemBarsBehavior =
+                WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+            hide(WindowInsetsCompat.Type.systemBars())
+        }
+    }
+    onDispose {
+        context.getActivity()?.window?.let { window ->
+            if (currentRotationAnimation != null) {
+                window.attributes = window.attributes.apply {
+                    rotationAnimation = WindowManager.LayoutParams.ROTATION_ANIMATION_ROTATE
```

Comment:
I've modified this to now restore the rotationAnimation to the old rotation animation before the DisposableEffect was applied. This seems to work pretty well, but I am not quite sure using a DisposableEffect is the right thing to do here. Based on [this documentation](https://developer.android.com/develop/ui/compose/layouts/insets#insets-and-phases) it's possible it may not apply correctly since we are getting the window during composition.

I'll continue to look at this to see if there is a better way using Compose's existing inset modifiers.